engine/cdn/cdn.go

```diff
@@ -0,0 +1,160 @@
+package cdn
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/gorilla/mux"
+
+	"github.com/ovh/cds/engine/api"
+	"github.com/ovh/cds/engine/api/cache"
+	"github.com/ovh/cds/engine/api/services"
+	"github.com/ovh/cds/engine/cdn/objectstore"
+	"github.com/ovh/cds/sdk"
+	"github.com/ovh/cds/sdk/cdsclient"
+	"github.com/ovh/cds/sdk/log"
+)
+
+const defaultBackendName = "default"
+
+// New returns a new service
+func New() *Service {
+	s := new(Service)
+	s.Router = &api.Router{
+		Mux: mux.NewRouter(),
+	}
+	return s
+}
+
+func (s *Service) Init(config interface{}) (cdsclient.ServiceConfig, error) {
+	var cfg cdsclient.ServiceConfig
+	sConfig, ok := config.(Configuration)
+	if !ok {
+		return cfg, sdk.WithStack(fmt.Errorf("invalid CDN service configuration"))
+	}
+
+	cfg.Host = sConfig.API.HTTP.URL
+	cfg.Token = sConfig.API.Token
+	cfg.InsecureSkipVerifyTLS = sConfig.API.HTTP.Insecure
+	cfg.RequestSecondsTimeout = sConfig.API.RequestTimeout
+	return cfg, nil
+}
+
+// ApplyConfiguration apply an object of type CDN.Configuration after checking it
+func (s *Service) ApplyConfiguration(config interface{}) error {
+	if err := s.CheckConfiguration(config); err != nil {
+		return err
+	}
+	var ok bool
+	s.Cfg, ok = config.(Configuration)
+	if !ok {
+		return fmt.Errorf("Invalid configuration")
+	}
+
+	s.ServiceName = s.Cfg.Name
+	s.ServiceType = services.TypeCDN
+	s.HTTPURL = s.Cfg.URL
+	s.MaxHeartbeatFailures = s.Cfg.API.MaxHeartbeatFailures
+	return nil
+}
+
+// CheckConfiguration checks the validity of the configuration object
+func (s *Service) CheckConfiguration(config interface{}) error {
+	sConfig, ok := config.(Configuration)
+	if !ok {
+		return fmt.Errorf("Invalid configuration")
+	}
+
+	if sConfig.URL == "" {
+		return fmt.Errorf("your CDS configuration seems to be empty. Please use environment variables, file or Consul to set your configuration")
+	}
+	if sConfig.Name == "" {
+		return fmt.Errorf("please enter a name in your CDN configuration")
+	}
+
+	return nil
+}
+
+// Serve will start the http api server
+func (s *Service) Serve(c context.Context) error {
+	ctx, cancel := context.WithCancel(c)
+	defer cancel()
+
+	//Init the cache
+	var errCache error
+	s.Cache, errCache = cache.New(s.Cfg.Cache.Redis.Host, s.Cfg.Cache.Redis.Password, s.Cfg.Cache.TTL)
+	if errCache != nil {
+		return fmt.Errorf("Cannot connect to redis instance : %v", errCache)
+	}
+
+	// init drivers
+	if err := s.initDefaultDrivers(ctx); err != nil {
+		return sdk.WrapError(err, "cannot init drivers")
+	}
+
+	//Init the http server
+	s.initRouter(ctx)
+	server := &http.Server{
+		Addr:           fmt.Sprintf("%s:%d", s.Cfg.HTTP.Addr, s.Cfg.HTTP.Port),
+		Handler:        s.Router.Mux,
+		ReadTimeout:    10 * time.Minute,
```

Comment:
For a content de livery service, it is weird to have a readTimeout and writeTimeout